packages/google_sign_in/lib/google_sign_in.dart

```diff
@@ -55,15 +63,30 @@ class GoogleSignInAccount implements GoogleIdentity {
   final String _idToken;
   final GoogleSignIn _googleSignIn;
 
-  Future<GoogleSignInAuthentication> get authentication async {
+  /// Retrieve [GoogleSignInAuthentication] for this account.
+  ///
+  /// [shouldRecoverAuth] sets whether to attempt to recover authentication if
+  /// user action is needed. If an attempt to recover authentication fails a
+  /// [PlatformException] is thrown with possible error code
+  /// [kFailedToRecoverAuthError].
+  ///
+  /// Otherwise, if [shouldRecoverAuth] is false and the authentication can be
+  /// recovered by user action a [PlatformException] is thrown with error code
+  /// [kUserRecoverableAuthError].
+  Future<GoogleSignInAuthentication> getAuthentication({
+    bool shouldRecoverAuth = true,
```

Comment:
Can we set this as a property of the Account (or even a property of the GoogleSignIn API itself?)

That way it is not a breaking change. Otherwise, I assume you also need to make authHeaders a function that takes this parameter. Most apps do not deal with tokens directly. They just fetch the headers.

Reply:
I confirmed that Mulligan is OK with this. They don't care about this Dart interface since they don't use it internally.

I now propose that we don't make a change to the Dart interface at all. We just pass in `true` as shouldRecoverAuth in here.